### PR TITLE
En 1606 improve seo page configuration

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.entando.entando.plugins.jpseo.aps.system.JpseoSystemConstants;
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.Metatag;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.PageMetatag;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.SeoPageExtraConfigDOM;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.SeoPageMetadata;
+
+public class SeoMetatagsTag extends OutSupport {
+
+    private static final Logger _logger = LoggerFactory.getLogger(SeoMetatagsTag.class);
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: No underscores.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    
+    private String var;
+
+    @Override
+    public int doEndTag() throws JspException {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: Add a comment on the function here. Pretty complex method. need some clarity on how/what it is doing.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            this.evalValue(output.toString());
+        } catch (Throwable t) {
+            _logger.error("error in doStartTag", t);
+            throw new JspException("Error during tag initialization ", t);
+        }
+        this.release();
+        return EVAL_PAGE;
+    }
+
+    protected void evalValue(String output) throws JspException {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: Add small comment.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/global-messages_en.properties [hunk 4]
```diff
@@ -12,10 +14,10 @@ jpseo.label.config=Seo configuration
 jpseo.label.example=Eg
 
 jpseo.label.friendlyCode=Friendly code
-jpseo.label.pageDescription=Description
+jpseo.label.pageDescription=SEO Description
 jpseo.label.useBetterDescriptions=When available, use extra description in search.
-jpseo.label.xmlConfig=Complex parameters
-xmlConfig=Complex parameters
+jpseo.label.inheritFromDefaultLang=Inherit from default lang
+jpseo.label.addMetatags=Add Metatag
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/global-messages_en.properties]: What does this param jpseo.label.addMetatags do?

Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/global-messages_en.properties]: "jpseo.label.addMetatags" the key of the label of the "Add metatag" section in the page edit UI

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java (hunk 3 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: done!

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java (hunk 2 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: Added javadoc in the header of the class

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java (hunk 1 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: Fixed

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+ * @author E.Santoboni
+ */
+public class SeoMetatagsTag extends OutSupport {
+
+    private static final Logger logger = LoggerFactory.getLogger(SeoMetatagsTag.class);
+    
+    private String var;
+
+    @Override
+    public int doEndTag() throws JspException {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/tags/SeoMetatagsTag.java]: Can this be unit tested? I don't see test coverage for this.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java [hunk 6]
```diff
@@ -105,56 +98,48 @@ public void executeExtraValueFormForEdit(JoinPoint joinPoint) {
...
             request.setAttribute(PARAM_FRIENDLY_CODE, pageMetadata.getFriendlyCode());
             request.setAttribute(PARAM_USE_EXTRA_DESCRIPTIONS, pageMetadata.isUseExtraDescriptions());
-            request.setAttribute(PARAM_XML_CONFIG, pageMetadata.getXmlConfig());
             ApsProperties props = pageMetadata.getDescriptions();
             if (null != props) {
-                Iterator<Object> it = props.keySet().iterator();
-                while (it.hasNext()) {
-                    String key = (String) it.next();
-                    request.setAttribute(PARAM_DESCRIPTION_PREFIX + key, props.get(key));
+                Iterator<Object> iter = props.keySet().iterator();
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java]: Does this have test coverage?

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java [hunk 7]
```diff
@@ -256,15 +198,28 @@ protected IPage buildPageMetadataForUpdate(HttpServletRequest request, PageActio
...
         String friendlyCode = request.getParameter(PARAM_FRIENDLY_CODE);
-        String xmlConfig = request.getParameter(PARAM_XML_CONFIG);
         Iterator<Lang> langsIter = this.getLangManager().getLangs().iterator();
         while (langsIter.hasNext()) {
             Lang lang = (Lang) langsIter.next();
             String titleKey = PARAM_DESCRIPTION_PREFIX + lang.getCode();
             String title = request.getParameter(titleKey);
             if (null != title) {
-                descriptions.put(lang.getCode(), title.trim());
+                PageMetatag meta = new PageMetatag(lang.getCode(), "description", title.trim());
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java]: Does this have test coverage

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageAction.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.Metatag;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.PageMetatag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.IMetatagCatalog;
+
+/**
+ * @author E.Santoboni
+ */
+public class SeoPageAction extends PageAction {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageAction.java]: Missing test coverage for this whole class

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageActionUtils.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.lang3.StringUtils;
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.Metatag;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.PageMetatag;
+
+public class SeoPageActionUtils {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageActionUtils.java]: Missing test coverage for this class

File: plugins/entando-plugin-jpseo/src/main/webapp/WEB-INF/plugins/jpseo/apsadmin/jsp/portal/entryPageLang-jack.jsp [hunk 10]
```diff
@@ -0,0 +1,190 @@
+<%@ taglib prefix="s" uri="/struts-tags" %>
+<%@ taglib uri="/aps-core" prefix="wp" %>
+<%@ taglib uri="/apsadmin-core" prefix="wpsa" %>
+<%@ taglib uri="/apsadmin-form" prefix="wpsf" %>
+
+<s:set var="mykey" value="'description_lang_'+#lang.code" />
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/webapp/WEB-INF/plugins/jpseo/apsadmin/jsp/portal/entryPageLang-jack.jsp]: Is there any way to test these JSPs? Lots of logic in here.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java (hunk 6 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java]: In test class TestPageAction (in test package) I have tested also the aspect class

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java (hunk 7 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/PageActionAspect.java]: In test class TestPageAction (in test package)

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageAction.java (hunk 8 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageAction.java]: In test class TestPageAction (in test package)

**Pull request merged**